Libraries/LibWeb/Parser/HTMLDocumentParser.cpp

```diff
@@ -1798,7 +1805,53 @@ void HTMLDocumentParser::handle_in_table(HTMLToken& token)
         reset_the_insertion_mode_appropriately();
         return;
     }
-    TODO();
+    if (token.is_end_tag() && token.tag_name().is_one_of(HTML::TagNames::body, HTML::TagNames::caption, HTML::TagNames::col, HTML::TagNames::colgroup, HTML::TagNames::html, HTML::TagNames::tbody, HTML::TagNames::td, HTML::TagNames::tfoot, HTML::TagNames::th, HTML::TagNames::thead, HTML::TagNames::tr)) {
+        PARSE_ERROR();
+        return;
+    }
+    if ((token.is_start_tag() && token.tag_name().is_one_of(HTML::TagNames::style, HTML::TagNames::script, HTML::TagNames::template_))
+         || (token.is_end_tag() && token.tag_name() == HTML::TagNames::template_)) {
+        process_using_the_rules_for(InsertionMode::InHead, token);
+        return;
+    }
+    if (token.is_start_tag() && token.tag_name() == HTML::TagNames::input) {
+        auto type_attribute = token.attribute(HTML::AttributeNames::type);
+        if (type_attribute.is_null() || !type_attribute.equals_ignoring_case("hidden")) {
+            goto AnythingElse;
+        }
+
+        PARSE_ERROR();
+        insert_html_element(token);
+
+        // FIXME: Is this the correct interpretation of "Pop that input element off the stack of open elements."?
+        //        Because this wording is the first time it's seen in the spec.
+        //        Other times it's worded as: "Immediately pop the current node off the stack of open elements."
```

Comment:
That's how I interpret it FWIW. But yeah, it's annoying the way the spec sometimes subtly changes the wording like this. Makes you feel like they're implying something different.